# HG changeset patch
# User Stefan Mahr <dac922@gmx.de>
# Date 1394972365 -3600
#      Sun Mar 16 13:19:25 2014 +0100
# Node ID a18e223224c516ba6aca1cca601e3bc586f94ac3
# Parent  ec55f6870efb550e590197b504e968668576c0e2
Fix pkg install -global (bug #41644)

* scripts/pkg/private/default_prefix.m (default_prefix):
  Add optional second input argument 'desc'

* scripts/pkg/private/getarchprefix.m (getarchprefix):
  Call 'default_prefix' with 'desc' as second parameter

diff --git a/scripts/pkg/private/default_prefix.m b/scripts/pkg/private/default_prefix.m
--- a/scripts/pkg/private/default_prefix.m
+++ b/scripts/pkg/private/default_prefix.m
@@ -17,19 +17,24 @@
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {[@var{prefix}, @var{archprefix} =} @
 ## default_prefix (@var{global_install})
 ## Undocumented internal function.
 ## @end deftypefn
 
-function [prefix, archprefix] = default_prefix (global_install)
+function [prefix, archprefix] = default_prefix (global_install, desc)
   if (global_install)
     prefix = fullfile (OCTAVE_HOME (), "share", "octave", "packages");
-    archprefix = fullfile (octave_config_info ("libdir"), "octave",
-                           "packages", [desc.name "-" desc.version]);
+    if (nargin == 2)
+      archprefix = fullfile (octave_config_info ("libdir"), "octave",
+                             "packages", [desc.name "-" desc.version]);
+    else
+      archprefix = fullfile (octave_config_info ("libdir"), "octave",
+                             "packages");
+    endif
   else
     prefix = tilde_expand (fullfile ("~", "octave"));
     archprefix = prefix;
   endif
 endfunction
 
diff --git a/scripts/pkg/private/getarchprefix.m b/scripts/pkg/private/getarchprefix.m
--- a/scripts/pkg/private/getarchprefix.m
+++ b/scripts/pkg/private/getarchprefix.m
@@ -19,14 +19,14 @@
 
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {[@var{archprefix} =} getarchprefix (@var{desc}, @var{global_install})
 ## Undocumented internal function.
 ## @end deftypefn
 
 function archprefix = getarchprefix (desc, global_install)
   if (global_install)
-    [~, archprefix] = default_prefix (global_install);
+    [~, archprefix] = default_prefix (global_install, desc);
   else
     archprefix = desc.dir;
   endif
 endfunction
 
